From 980d0444de6c0a68aa520fd4cd1f12dc407bf6a8 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@renesas.com>
Date: Tue, 8 Mar 2022 14:38:38 +0700
Subject: [PATCH 3/9] Do not parse and init lossy mem area for G2L/G3E/V2H/V2N
 Series

RZ/G2L, RZ/G3E, RZ/V2H and RZ/V2N series do not support lossy mem area.
So this commit is created to let mmngr not parse and init
lossy mem area for them.

Signed-off-by: trungvanle <trung.le.xk@renesas.com>
---
 .../mmngr-module/files/mmngr/drv/mmngr_drv.c  | 42 ++++++++++++-------
 1 file changed, 28 insertions(+), 14 deletions(-)

diff --git a/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c b/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c
index 5d31aa6..7dee970 100644
--- a/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c
+++ b/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c
@@ -1,7 +1,7 @@
 /*************************************************************************/ /*
  MMNGR
 
- Copyright (C) 2015-2019 Renesas Electronics Corporation
+ Copyright (C) 2015-2025 Renesas Electronics Corporation
 
  License        Dual MIT/GPLv2
 
@@ -1255,6 +1255,15 @@ static int parse_legacy_memory_node(struct device_node *np)
 }
 #endif
 
+static const struct soc_device_attribute rz_matches[] = {
+	{ .family = "RZ/G2L" },
+	{ .family = "RZ/G2UL" },
+	{ .family = "RZ/G3E" },
+	{ .family = "RZ/V2H" },
+	{ .family = "RZ/V2N" },
+	{ /* sentinel*/ }
+};
+
 static int parse_reserved_mem_dt(struct device_node *np)
 {
 	int ret = 0;
@@ -1269,15 +1278,18 @@ static int parse_reserved_mem_dt(struct device_node *np)
 			 "(linux,multimedia) from DT\n");
 		return ret;
 	}
-	/* Parse reserved memory for lossy compression feature */
-	ret = _parse_reserved_mem_dt(np, "memory-region",
-				     "linux,lossy_decompress",
-				     &mm_lossybuf_addr,
-				     &mm_lossybuf_size);
-	if (ret) {
-		pr_warn("Failed to parse Lossy reserved area" \
-			"(linux,lossy_decompress) from DT\n");
-		ret = 0; /* Let MMNGR support other features */
+
+	if (!soc_device_match(rz_matches)) {
+		/* Parse reserved memory for lossy compression feature */
+		ret = _parse_reserved_mem_dt(np, "memory-region",
+					     "linux,lossy_decompress",
+					     &mm_lossybuf_addr,
+					     &mm_lossybuf_size);
+		if (ret) {
+			pr_warn("Failed to parse Lossy reserved area" \
+				"(linux,lossy_decompress) from DT\n");
+			ret = 0; /* Let MMNGR support other features */
+		}
 	}
 	return ret;
 }
@@ -1787,10 +1799,12 @@ static int mm_probe(struct platform_device *pdev)
 	}
 #endif
 
-	ret = init_lossy_info();
-	if (ret) {
-		pr_err("MMD mm_init ERROR\n");
-		return -1;
+	if (!soc_device_match(rz_matches)) {
+		ret = init_lossy_info();
+		if (ret) {
+			pr_err("MMD mm_init ERROR\n");
+			return -1;
+		}
 	}
 
 	p = kzalloc(sizeof(struct MM_DRVDATA), GFP_KERNEL);
-- 
2.25.1

